kong/cmd/utils/prefix_handler.lua

```diff
@@ -378,6 +378,60 @@ local function write_process_secrets_file(path, data)
   return true
 end
 
+local function pre_create_lmdb(conf)
+  local prefix = conf.prefix
+  local lmdb_path = conf.lmdb_environment_path or "dbless.lmdb"
+
+  local dir_name
+  if lmdb_path:sub(1, 1) == "/" then
+    dir_name = lmdb_path
+
+  else
+    dir_name = prefix .. "/" .. lmdb_path
+  end
+
+  if pl_path.isdir(dir_name) then
+    return true
+  end
+
+  local user = string.match(conf.nginx_user or "", "%w+")
+
+  if not user then
+    return nil, "nginx_user is not set in Kong config, "..
+                "skipping verification of LMDB file permissions"
+  end
+
+  log.debug("LMDB directory '%s' does not exist, " ..
+            "pre-creating with the correct permissions",
+            dir_name)
+
+  local ok, err = pl_path.mkdir(dir_name)
+  if not ok then
+    return nil, "can not create directory for LMDB " .. dir_name ..
+                ", err: " .. err
+  end
+
+  local cmds = {
+    string.format("chown %s %s && chmod 0700 %s",
```

Comment:
The group information from `nginx_user` is important, we should `chown user:group path` here.

Reply:
done